src/credentials/GroupDataProviderImpl.h

```diff
@@ -208,7 +225,7 @@ class GroupDataProviderImpl : public GroupDataProvider
     class GroupSessionIteratorImpl : public GroupSessionIterator
     {
     public:
-        GroupSessionIteratorImpl(GroupDataProviderImpl & provider, uint16_t session_id);
+        GroupSessionIteratorImpl(GroupDataProviderImpl & provider, Crypto::SessionKeystore & keystore, uint16_t session_id);
```

Comment:
`GroupDataProviderImpl` (which is a known concrete class here) could have the SessionKeystore getter as private. Then the inner classes could just call the getter on the provider, rather than needing to pass it down. Would reduce some of this boilerplate.